src/appleseed/renderer/kernel/rendering/pixelrendererbase.cpp

```diff
@@ -51,22 +54,66 @@ PixelRendererBase::PixelRendererBase()
 {
 }
 
+PixelRendererBase::PixelRendererBase(
+    const Frame&        frame,
+    const size_t        thread_index)
+  : m_invalid_pixel_count(0)
+{
+    m_invalid_sample_aov_index = frame.create_extra_aov_image("invalid_samples");
```

Comment:
The invalid samples AOV should be optional (since most of the time we don't need it, but it uses significant amounts of memory).

Reply:
@dictoon Is this possible to share some Parameters along pixel renderer base and sub classes ?

Reply:
Or, I can use Frame::save_extra_aovs ? And we could do the same in the adaptive renderer ? 
Because `use diagnostics` and `save extra aovs` work only if they are both enabled.

Reply:
I think this coupling would be difficult to understand. In fact, do we agree that we have the same issue with the adaptive pixel sampler's diagnostic AOVs? Even if they are enabled, they aren't save to disk unless Save Extra AOVs is checked in the frame? If so then let's do the same for the invalid samples AOV, for now. (By the way, a checkbox must be added to the render settings to let users enable the invalid sample AOV.)

Reply:
Ok I'll do that !